backend/src/Controller/Subscription/SubscriptionController.php

```diff
@@ -0,0 +1,248 @@
+     * @Security(name="Bearer")
+     */
+    public function createSubscription(Request $request): JsonResponse
+    {
+        $data = json_decode($request->getContent(), true);
+
+        $request = $this->autoMapping->map(stdClass::class, SubscriptionCreateRequest::class, (object)$data);
+
+        $request->setStoreOwner($this->getUserId());
+
+        $violations = $this->validator->validate($request);
+
+        if (\count($violations) > 0) {
+            $violationsString = (string) $violations;
+
+            return new JsonResponse($violationsString, Response::HTTP_OK);
+        }
+
+        $result = $this->subscriptionService->createSubscription($request);
+
+        return $this->response($result, self::CREATE);
+    }
+    
+    /**
+     * @Route("nextsubscription", name="nextSubscription", methods={"POST"})
+     * @IsGranted("ROLE_OWNER")
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function nextSubscription(Request $request)
+    {
+        $data = json_decode($request->getContent(), true);
+
+        $request = $this->autoMapping->map(stdClass::class, SubscriptionNextRequest::class, (object)$data);
+
+        $request->setOwnerID($this->getUserId());
+
+        // $violations = $this->validator->validate($request);
+
+        // if (\count($violations) > 0) {
+        //     $violationsString = (string) $violations;
+
+        //     return new JsonResponse($violationsString, Response::HTTP_OK);
+        // }
+
+        $result = $this->subscriptionService->nextSubscription($request);
+
+        return $this->response($result, self::CREATE);
+    }
+
+    /**
+     * store:get subscriptions to the store owner with the identification of the current subscription.
+     * @Route("subscriptionsforstoreowner", name="getSubscriptionsForStoreOwner", methods={"GET"})
+     * @IsGranted("ROLE_OWNER")
+     * @return JsonResponse
+     *
+     * @OA\Tag(name="Subscription")
+     *
+     * @OA\Parameter(
+     *      name="token",
+     *      in="header",
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\Response(
+     *      response=200,
+     *      description="Returns subscriptions",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="array", property="Data",
+     *            @OA\Items(
+     *                @OA\Property(type="integer", property="id"),
+     *                @OA\Property(type="string", property="packageName"),
+     *                @OA\Property(type="object", property="startDate"),
+     *                @OA\Property(type="object", property="endDate"),
+     *                @OA\Property(type="string", property="status"),
+     *                @OA\Property(type="string", property="note"),
+     *                @OA\Property(type="string", property="isCurrent"),
```

Comment:
I think we need to change the type of isCurrent, in the annotation, from string to boolean.

Reply:
no, is string.